contrib/codegen/templates/input/input_snapshot_manual_builder.gotmpl

```diff
@@ -0,0 +1,76 @@
+{{ $groups := imported_groups }}
+
+/*
+	Utility for manually building input snapshots. Used primarily in tests.
+*/
+package {{ package }}
+
+import (
+multiclusterv1alpha1 "github.com/solo-io/skv2/pkg/api/multicluster.solo.io/v1alpha1"
+multiclusterv1alpha1sets "github.com/solo-io/skv2/pkg/api/multicluster.solo.io/v1alpha1/sets"
+
+{{- range $group := $groups }}
+	{{ $import_prefix := group_import_name $group }}
+	{{ $client_import_prefix := (printf "%v_clients" (group_import_name $group)) }}
+	{{ $set_import_prefix := (printf "%v_sets" (group_import_name $group)) }}
+	{{ $import_prefix }} "{{ group_import_path $group }}"
+	{{ $client_import_prefix }} "{{ client_import_path $group }}"
+	{{ $set_import_prefix }} "{{ set_import_path $group }}"
+{{- end }}
+
+)
+
+type InputSnapshotManualBuilder struct {
+name string
+{{- range $group := $groups }}
+	{{ $set_import_prefix := printf "%v_sets" (group_import_name $group) }}
+	{{- range $resource := $group.Resources }}
+		{{- $kindLowerCamel := lower_camel $resource.Kind }}
+		{{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
+		{{ $kindLowerCamelPlural }} {{ $set_import_prefix }}.{{ $resource.Kind }}Set
+	{{- end }}
+{{- end }}
+
+}
+
+func NewInputSnapshotManualBuilder(name string) *InputSnapshotManualBuilder {
+return &InputSnapshotManualBuilder{
+name:               name,
+{{- range $group := $groups }}
+	{{ $set_import_prefix := printf "%v_sets" (group_import_name $group) }}
+	{{- range $resource := $group.Resources }}
+		{{- $kindPlural := pluralize $resource.Kind }}
+		{{- $kindLowerCamel := lower_camel $resource.Kind }}
+		{{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
+		{{ $kindLowerCamelPlural }}: {{ $set_import_prefix }}.New{{ $resource.Kind }}Set(),
+	{{- end }}
+{{- end }}
+}
+}
+
+func (i *InputSnapshotManualBuilder) Build() Snapshot {
+return NewSnapshot(
+i.name,
+{{- range $group := $groups }}
+	{{ $set_import_prefix := printf "%v_sets" (group_import_name $group) }}
+	{{- range $resource := $group.Resources }}
+		{{- $kindPlural := pluralize $resource.Kind }}
+		{{- $kindLowerCamel := lower_camel $resource.Kind }}
+		{{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
+		i.{{ $kindLowerCamelPlural }},
+	{{- end }}
+{{- end }}
+)
+}
+
+{{- range $group := $groups }}
+	{{- range $resource := $group.Resources }}
+		{{- $kindPlural := pluralize $resource.Kind }}
+		{{- $kindLowerCamel := lower_camel $resource.Kind }}
+		{{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
+		func (i *InputSnapshotManualBuilder) Add{{ $kindPlural }}({{ $kindLowerCamelPlural }} []*{{ group_import_name $group }}.{{ $resource.Kind }}) *InputSnapshotManualBuilder {
```

Comment:
Can we make these args variadic instead of a slice?
```suggestion
		func (i *InputSnapshotManualBuilder) Add{{ $kindPlural }}({{ $kindLowerCamelPlural }} ...*{{ group_import_name $group }}.{{ $resource.Kind }}) *InputSnapshotManualBuilder {
```